cmd/api/src/auth/model.go

```diff
@@ -90,43 +90,52 @@ type AuditLogger interface {
 	AppendAuditLog(ctx context.Context, entry model.AuditEntry) error
 }
 
-type Authorizer interface {
-	HasPermission(ctx Context, requiredPermission model.Permission, grantedPermissions model.Permissions) bool
-	AllowsPermission(ctx Context, requiredPermission model.Permission) bool
-	AllowsAllPermissions(ctx Context, requiredPermissions model.Permissions) bool
-	AllowsAtLeastOnePermission(ctx Context, requiredPermissions model.Permissions) bool
-	AuditLogUnauthorizedAccess(request *http.Request)
+type Authorizer struct {
+	auditLogger    AuditLogger
+	getPermissions GetPermissionsFunc
 }
 
-type authorizer struct {
-	auditLogger AuditLogger
+func NewCustomAuthorizer(auditLogger AuditLogger, getPermissionsFn GetPermissionsFunc) Authorizer {
+	if getPermissionsFn == nil {
+		getPermissionsFn = getPermissions
+	}
```

Comment:
Is this an expected scenario? I only see it used with a non nil func passed in so was curious

Reply:
Because it's accepting a fn, just covering our bases with the nil check.